strings/src/main/res/values/strings.xml

```diff
@@ -628,19 +628,45 @@
     <string name="form_update_error">Form update failed</string>
     <string name="form_update_succeeded">Form update succeeded</string>
 
+    <string name="load_remote_form_error">Form list download failed</string>
+
+    <string name="all_downloads_succeeded">All downloads succeeded!</string>
+    <string name="some_downloads_failed">%1$s of %2$s downloads failed!</string>
+    <string name="show_details">Show details</string>
+    <string name="form_details">ID: %1$s Version: %2$s</string>
+    <string name="failure">Failure</string>
+
+    <!--
+    ##############################################
+    # GET BLANK FORM - ERROR MESSAGES
+    ##############################################
+    -->
+
     <string name="report_to_project_lead">If you keep having this problem, report it to the person who asked you to collect data.</string>
+
+    <!-- Error message displayed when Collect can't connect to a server -->
     <string name="unreachable_error">Collect can\'t reach the server at %s. Did you enter the URL correctly?</string>
+
+    <!-- Error message displayed when Collect can't connect to a server because of a security issue -->
     <string name="security_error">Collect can\'t connect securely to the server at %s. Did you enter the URL correctly?</string>
+
+    <!-- Error message displayed when a server returns an error status code -->
     <string name="server_error">The server %1$s returned status code %2$s.</string>
+
+    <!-- Error message displayed when a server returns an invalid response body -->
     <string name="invalid_response">The server %s provided an invalid response.</string>
-    <string name="failure">Failure</string>
 
-    <string name="load_remote_form_error">Form list download failed</string>
+    <!-- Error message displayed when the server has not provided a hash identifier for the form being downloaded -->
+    <string name="form_with_no_hash_error">This form has no hash.</string>
+
+    <!-- Error message displayed when a form being downloaded cannot be parsed -->
+    <string name="form_parsing_error">This form cannot be parsed.</string>
+
+    <!-- Error message displayed when a disk error occurs during form download -->
+    <string name="form_save_disk_error">Disk error occurred while downloading this form.</string>
 
-    <string name="form_with_no_hash_error">The form has no hash.</string>
-    <string name="form_parsing_error">The form can not be parsed.</string>
-    <string name="form_save_disk_error">Disk exception occurred during saving the form.</string>
-    <string name="form_with_invalid_submission_error">The form has invalid submission.</string>
+    <!-- Error message displayed when a form being downloaded specifies an invalid submission URL -->
+    <string name="form_with_invalid_submission_error">This form has an invalid submission URL.</string>
```

Comment:
@yanokwa it'd be great if you could review these error messages for me. They will all have `report_to_project_lead` appended to the end of them (see `FormDownloadExceptionMapper` and `FormSourceExceptionMapper`) to try and push users to complain through their project/programme rather than leaving a negative review.

Reply:
I think we should add a bit of a hint that the project lead can use to track down the problem. Is it a form problem? A server problem?

```
The server did not provide a form hash.
This form has an invalid submission URL.
A device error occurred during download.
``

`form_parsing_error` is a little trickier. Why can't the form be parsed? Is it because the file is corrupt due to a network error? Is it because of poor form design? I like processed over parsed, but I don't feel strongly about it. 

We could also be a little less passive. "Your form has an invalid submission URL". "Your server did not provide a form has."

Reply:
Yeah I like the idea of being more specific about **where** the error occurred. I've made changes based on that feedback. I'm not sure about the slightly more aggressive "your" so I've not made that change.